pixel-saver@deadalnix.me/decoration.js

```diff
@@ -148,11 +159,13 @@ function getOriginalState(win) {
 		      (state ? '0x1' : '0x0')];
 		LOG(cmd.join(' '));
 		Util.spawn(cmd);
-		return win._pixelSaverOriginalState = state;
+		return win._pixelSaverOriginalState = state ? WindowState.HIDE_TITLEBAR : WindowState.DEFAULT;
 	}
 	
 	WARN("Can't find original state for " + win.title + " with id " + id);
-	return false;
+
+	// if we can't find an atom, assume the titlebar not to be hidden
+	return win._pixelSaverOriginalState = WindowState.DEFAULT;
```

Comment:
I'm no expert here. Looking at the source code, it seems GTK just sets the Atom when this feature is requested. Thus, the absence of this Atom should indicate that title bars are displayed. It might depend on other factors (theme), but as far as this logic goes all is normal. The detection finished without problems. When we set the Atom, we will remove it upon exit.

In contrast UNKNOWN indicates that the xprop call failed, and we don't know nothing about the state. No reset is performed upon exit.